src/components/Course/Course.vue

```diff
@@ -1,96 +1,109 @@
+        active: active,
       }"
+      class="course"
     >
-      <img src="@/assets/images/dots/sixDots.svg" alt="" />
-    </div>
-    <div class="course-content" @click="courseOnClick()">
-      <div class="course-main">
-        <div class="course-top">
-          <div class="course-left">
-            <div
-              class="course-code"
-              data-cyId="courseCode"
-              :style="{
-                color: compact ? '#3d3d3d' : '#858585',
-              }"
-            >
-              {{ courseObj.code }}
+      <save-course-modal
+        :courseCode="courseCode"
+        @close-save-course-modal="closeSaveCourseModal"
+        @save-course="saveCourse"
+        @add-collection="addCollection"
+        v-if="isSaveCourseOpen"
+      />
+      <edit-color
+        :editedColor="editedColor"
+        @color-course="colorCourse"
+        @color-subject="colorSubject"
+        @close-edit-color="closeEditColorModal"
+        v-if="isEditColorOpen"
+      />
+      <div
+        class="course-color"
+        :style="cssVars"
+        :class="{
+          'course-color--active': active,
+        }"
+      >
+        <img src="@/assets/images/dots/sixDots.svg" alt="" />
+      </div>
+      <div class="course-content" @click="courseOnClick()">
+        <div class="course-main">
+          <div class="course-top">
+            <div class="course-left">
+              <div
+                class="course-code"
+                data-cyId="courseCode"
+                :style="{
+                  color: compact ? '#3d3d3d' : '#858585',
+                }"
+              >
+                {{ courseObj.code }}
+              </div>
+              <course-caution
+                v-if="!isReqCourse && compact"
+                :course="courseObj"
+                :isCompactView="true"
+              />
             </div>
+            <button
+              v-if="!isReqCourse && isSemesterCourseCard"
+              class="course-dotRow"
+              @click="openMenu"
+            >
+              <img src="@/assets/images/dots/threeDots.svg" alt="open menu for course card" />
+            </button>
+            <button
+              v-else-if="!isReqCourse && !isSemesterCourseCard"
+              class="course-trash"
+              @click.stop="deleteCourseFromCollection"
+              @mouseover="hoverTrashIcon"
+              @mouseleave="unhoverTrashIcon"
+            >
+              <img :src="trashIcon" alt="delete course from collection" />
+            </button>
+          </div>
+          <div v-if="!compact" class="course-name">{{ courseObj.name }}</div>
+          <div v-if="!compact" class="course-info">
+            <span class="course-credits">{{ creditString }}</span>
+            <span v-if="semesterString" class="course-semesters">{{ semesterString }}</span>
             <course-caution
-              v-if="!isReqCourse && compact"
+              v-if="!isReqCourse && !isSchedGenCourse"
```

Comment:
Just for reference could you explain what this change was motivated by? Doesn't look immediately relevant to the notes feature.